envoy/http/header_validator.h

```diff
@@ -95,6 +95,21 @@ class HeaderValidator {
   using ResponseHeaderMapValidationResult = RejectResult;
   virtual ResponseHeaderMapValidationResult
   validateResponseHeaderMap(ResponseHeaderMap& header_map) PURE;
+
+  /**
+   * Validate the entire request trailer map.
+   * Returning the Reject value causes HTTP requests to be rejected with the 502 status,
```

Comment:
nit: should this be HTTP request to be rejected (no s) here and below?

Reply:
Done